src/shared/views/Login.js

```diff
@@ -310,6 +323,58 @@ class _Login extends React.Component<Props, State> {
             });
     };
 
+    OSMOauthCallback = (
+        osmCode: string,
+        osmState: string,
+        fbToken: string,
+        osmError: string,
+    ) => {
+        const { firebase, t } = this.props;
+        console.log('OSM callback', osmCode, osmState, fbToken, osmError);
+        // call /token now, which will get a token from firebase
+        // and redirect here via deeplink
+        if (fbToken === undefined) {
+            console.log('Fetching a token');
+            Linking.openURL(
+                `https://dev-mapswipe.web.app/token?code=${osmCode}&state=${osmState}`,
```

Comment:
Yes, same logic as your other comment about the prefix. It could be nice to deploy these functions under `mapswipe.org` or maybe a subdomain like `auth.mapswipe.org`, just to make it look a bit cleaner. I don't kow how much control we have over the domain though.